apps/src/applab/applab.js

```diff
@@ -782,7 +784,14 @@ Applab.init = function (config) {
         });
       }
     });
-
+  // Only send PROJECT_ACTIVITY event for students or non-authenticated users,
+  // AND only if they're on a project level.
+  if (
+    getStore().getState().currentUser?.userType !== 'teacher' &&
```

Comment:
There are a lot of the `getStore().getState().currentUser?.userType !== 'teacher'` repeated all over. I could abstract this into a function, but I'm not sure where to put it. Some kind of utils or shared package.